src/etwlogger_driver.c

```diff
@@ -277,50 +145,61 @@ static void perform_EventWriteLogVerboseEvent(const char* message, const char* f
     );
 
 #if CALL_CONSOLE_LOGGER
-    consolelogger_log(AZ_LOG_VERBOSE, NULL, NULL, 0, LOG_LINE, "%s", message);
+    consolelogger_log(AZ_LOG_VERBOSE, file, func, line, LOG_LINE, "%s", message);
 #endif
 }
 
+
+static const char vsnprintf_failure_message[] = "failure in vsnprintf";
+static const char FormatMessageA_failure_message[] = "failure in FormatMessageA";
 void etwlogger_log_with_GetLastError(const char* file, const char* func, int line, const char* format, ...)
 {
     DWORD lastError;
-    char* lastErrorAsString;
 
     lastError = GetLastError();
     lazyRegisterEventProvider();
 
     va_list args;
     va_start(args, format);
 
-    lastErrorAsString = lastErrorToString(lastError);
-    if (lastErrorAsString == NULL)
-    {
-        char* userMessage = vprintf_alloc(format, args);
-        if (userMessage == NULL)
+    char message[LOG_SIZE_REGULAR * (LOG_SIZE_REGULAR>=sizeof(vsnprintf_failure_message))]; /*this construct will generate a compile time error (array of size 0) when LOG_SIZE_REGULAR is not enough to hold even the failure message*/
```

Comment:
>(LOG_SIZE_REGULAR>=sizeof(vsnprintf_failure_message) [](start = 36, length = 52)

WTB more explicit static_assert macro (exists as built-in in C11 I guess but would be nice to have)

Reply:
Microsoft's C has documented _STATIC_ASSERT([https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/static-assert-macro?view=vs-2019](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/static-assert-macro?view=vs-2019)) which seems to do the same, without providing a string literal like C++ since C++17.

_STATIC_ASSERT comes from from <crtdbg.h>.

I would like to keep it as is, as this solution is more portable (doesn't involve a specific header) than _STATIC_ASSERT which is Microsoft's own implementation. Also is rather canon in  C world to have an array of size 0 when something goes wrong.

---
In reply to: [453751150](https://github.com/Azure/azure-c-logging/pull/11#discussion_r453751150) [](ancestors = 453751150)